naga/src/back/spv/block.rs

```diff
@@ -1069,11 +1086,52 @@ impl<'w> BlockContext<'w> {
                         arg3_id,
                     )),
                     Mf::FindLsb => MathOp::Ext(spirv::GLOp::FindILsb),
-                    Mf::FindMsb => MathOp::Ext(match arg_scalar_kind {
-                        Some(crate::ScalarKind::Uint) => spirv::GLOp::FindUMsb,
-                        Some(crate::ScalarKind::Sint) => spirv::GLOp::FindSMsb,
-                        other => unimplemented!("Unexpected findMSB({:?})", other),
-                    }),
+                    Mf::FindMsb => {
+                        if arg_ty.scalar_width() == Some(32) {
+                            let thing = match arg_scalar_kind {
+                                Some(crate::ScalarKind::Uint) => spirv::GLOp::FindUMsb,
+                                Some(crate::ScalarKind::Sint) => spirv::GLOp::FindSMsb,
+                                other => unimplemented!("Unexpected findMSB({:?})", other),
+                            };
+                            MathOp::Ext(thing)
+                        } else {
+                            // This polyfill doesn't work yet https://github.com/gfx-rs/wgpu/issues/5276
+                            let int_type_id = match *arg_ty {
+                                crate::TypeInner::Vector { size, scalar } => self.get_type_id(
+                                    LocalType::Value {
+                                        vector_size: Some(size),
+                                        scalar,
+                                        pointer_space: None,
+                                    }
+                                    .into(),
+                                ),
+                                crate::TypeInner::Scalar(scalar) => {
+                                    self.get_type_id(LookupType::Local(LocalType::Value {
+                                        vector_size: None,
+                                        scalar,
+                                        pointer_space: None,
+                                    }))
+                                }
+                                _ => unreachable!(),
+                            };
+
+                            let rev_id = self.gen_id();
+                            block.body.push(Instruction::unary(
+                                spirv::Op::BitReverse,
+                                int_type_id,
+                                rev_id,
+                                arg0_id,
+                            ));
+
+                            MathOp::Custom(Instruction::ext_inst(
+                                self.writer.gl450_ext_inst_id,
+                                spirv::GLOp::FindILsb,
+                                result_type_id,
+                                id,
+                                &[rev_id],
+                            ))
```

Comment:
I don't think we should add non-working code. Same above.

Reply:
Fair, I'll remove it later tonight.